provisioner/provisioner.go

```diff
@@ -563,16 +599,16 @@ func (p *freenasProvisioner) Provision(options controller.VolumeOptions) (*v1.Pe
 			VolumeMode: options.PVC.Spec.VolumeMode,
 			PersistentVolumeSource: v1.PersistentVolumeSource{
 				ISCSI: &v1.ISCSIPersistentVolumeSource{
-					TargetPortal:   config.ProvisionerTargetPortal,
-					Portals:        portals,
-					IQN:            iscsiConfig.Basename + ":" + iscsiName,
-					ISCSIInterface: config.ProvisionerISCSIInterface,
-					Lun:            int32(*targetToExtent.Lunid),
-					ReadOnly:       extent.Ro,
-					FSType:         config.FSType,
-					//DiscoveryCHAPAuth: false,
-					//SessionCHAPAuth:   false,
-					//SecretRef:         getSecretRef(getBool(options.Parameters["chapAuthDiscovery"]), getBool(options.Parameters["chapAuthSession"]), &v1.SecretReference{Name: viper.GetString("provisioner-name") + "-chap-secret"}),
+					TargetPortal:      config.ProvisionerTargetPortal,
+					Portals:           portals,
+					IQN:               iscsiConfig.Basename + ":" + iscsiName,
+					ISCSIInterface:    config.ProvisionerISCSIInterface,
+					Lun:               int32(*targetToExtent.Lunid),
+					ReadOnly:          extent.Ro,
+					FSType:            config.FSType,
+					DiscoveryCHAPAuth: config.DiscoveryCHAPAuth,
+					SessionCHAPAuth:   config.SessionCHAPAuth,
+					SecretRef:         config.AuthSecretRef,
```

Comment:
I presume this value is `nil` in the scenario where everything is off? Did you happen to run through some tests with auth disabled?

Reply:
I've done manual testing, and it worked with nil SecretRef. Basically, that is the default value for that field I presume.